### PR TITLE
8286473: Drop --enable-preview from Record related tests

### PR DESCRIPTION
**Pull request by shipilev**
There are plenty of tests failing on many architectures due to `--enable-preview` VM code introduced by Loom. This improvements eliminates some of the redundant `--enable-preview` clauses from the Record tests, since Records have been graduated from preview in JDK 16. 

Additional testing:
 - [x] Linux x86_64 fastdebug, affected tests still pass
 - [x]  Linux x86_32 fastdebug, affected tests start to pass

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286473](https://bugs.openjdk.java.net/browse/JDK-8286473): Drop --enable-preview from Record related tests

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.java.net/census#jpai) (@jaikiran - Committer)
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8626/head:pull/8626` \
`$ git checkout pull/8626`

Update a local copy of the PR: \
`$ git checkout pull/8626` \
`$ git pull https://git.openjdk.java.net/jdk pull/8626/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8626`

View PR using the GUI difftool: \
`$ git pr show -t 8626`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8626.diff">https://git.openjdk.java.net/jdk/pull/8626.diff</a>

</details>

### CODE REVIEW

**Comment by jaikiran** [comment]
Hello Aleksey, the `unreflect` test package has just one test and that one uses the `record` feature. So removing the `TEST.properties` for that entire package looks fine.

The changes to the other 2 tests look fine too, since like you note, they are just `record` related tests and don't have any other preview feature being tested. I think these two tests would need a copyright year update.

File: test/jdk/java/nio/Buffer/BulkPutBuffer.java [hunk 1]
```diff
@@ -51,8 +51,8 @@
  * @test
  * @bug 8219014 8245121
  * @summary Ensure that a bulk put of a buffer into another is correct.
- * @compile --enable-preview -source ${jdk.version} BulkPutBuffer.java
- * @run testng/othervm --enable-preview BulkPutBuffer
+ * @compile -source ${jdk.version} BulkPutBuffer.java
+ * @run testng/othervm BulkPutBuffer
  */
 public class BulkPutBuffer {
```
Comment (AlanBateman) [inline test/jdk/java/nio/Buffer/BulkPutBuffer.java:57]: You can drop the -source option from these tests too.

Comment (shipilev) [reply test/jdk/java/nio/Buffer/BulkPutBuffer.java:57]: Done.

**Review by AlanBateman — APPROVED** [review]

**Review by mlchung — APPROVED** [review]

**Comment by shipilev** [comment]
Thanks for reviews. I see GHA are mostly clean, with the usual x86_32 failures.

/integrate

**Pull request closed without merging**